packages/web/components/token-select.tsx

```diff
@@ -0,0 +1,288 @@
+import React, {
+  FunctionComponent,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import Image from "next/image";
+import { AppCurrency } from "@keplr-wallet/types";
+import { observer } from "mobx-react-lite";
+import { useStore } from "../stores";
+import classNames from "classnames";
+import { useBooleanWithWindowEvent } from "../hooks";
+
+/**
+ * TokenSelect's dropdown is attached to the nearest "relative" element.
+ */
+export const TokenSelect: FunctionComponent<{
```

Comment:
Have you seen my version of [Token Select](https://github.com/osmosis-labs/osmosis-frontend/blob/jonator/refactor/mobile-foundation/packages/web/components/control/token-select.tsx)  on the `/mobile-foundation` branch? It provides the option for showing balances via `CoinPretty` objects. Also, it uses `useFilteredData` to cleanly operate the search bar.

Reply:
Would take a look.